photutils/detection/findstars.py

```diff
@@ -7,18 +7,20 @@
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+class StarFinder(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __call__(self, data):
+        """Find potential stars in the given data."""
+        pass
 
-
-def daofind(data, threshold, fwhm, ratio=1.0, theta=0.0, sigma_radius=1.5,
-            sharplo=0.2, sharphi=1.0, roundlo=-1.0, roundhi=1.0, sky=0.0,
-            exclude_border=False):
+class DAOFind(StarFinder):
```

Comment:
Actually, thinking about this a bit more I realize now why they have this name: daophot has a task actually called just "find", and I think the IRAF star-finder is actually called "starfind".  

I still think my suggestion here makes sense - to call them "finder" rather than just "find", and note in the *docs* that they are based on the "FIND" and "starfind" tasks.  But I guess it kind of makes sense either way - in the current scheme it's basically "name of the algorithm as originally written", and in what I'm suggesting it's more "description of what the class is doing, but in a way that reminds you of the previous name"